apps/gitness/src/pages-v2/repo/stores/branch-selector-store.ts

```diff
@@ -0,0 +1,29 @@
+import { create } from 'zustand'
+
+import { BranchSelectorListItem, IBranchSelectorStore } from '@harnessio/ui/views'
+
+export const useBranchSelectorStore = create<IBranchSelectorStore>(set => ({
+  size: 'default',
+  setSize: size => set({ size }),
+
+  selectedBranch: undefined,
+  setSelectedBranch: (selectedBranch?: BranchSelectorListItem) => set({ selectedBranch }),
+
+  branchList: undefined,
+  setBranchList: (branchList?: BranchSelectorListItem[]) => set({ branchList }),
+
+  tagList: [{ name: '', sha: '' }],
+  setTagList: (tagList: BranchSelectorListItem[]) => set({ tagList }),
+
+  repoId: '',
+  setRepoId: (repoId: string) => set({ repoId }),
+
+  spaceId: '',
+  setSpaceId: (spaceId: string) => set({ spaceId }),
+
+  prefix: '',
+  setPrefix: (prefix: string | undefined) => set({ prefix }),
+
+  className: '',
+  setClassName: (className: string | undefined) => set({ className })
+}))
```

Comment:
why does `className` need to be a part of our store?

Reply:
Right, shouldn't be there. Let me pass it as a prop instead.

Reply:
Im confused why we need to pass a `className` prop at all?

Reply:
The button in Branch Selector is appending the className from props. I don't think the button will be styled differently specifically in a particular page, so we may remove it. Anyway, no page is passing className. It's an optional prop.

Reply:
Removed